### ISSUE TITLE
Seg fault in `dispatchee_registration_t::mark_ready()`

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4875]
Got this after about 5 minutes of running `table_fuzzer.py --servers 8 --threads 64 --serve-flags "--cache-size 100"` on `next` with a release mode binary.

```
2015-09-22T15:50:38.238761331 376.607789s error: Error in src/arch/runtime/thread_pool.cc at line 359:
2015-09-22T15:50:38.240173685 376.609201s error: Segmentation fault from reading the address 0x28.
2015-09-22T15:50:38.241654689 376.610682s error: Backtrace:
2015-09-22T15:50:42.376505458 380.745535s info: Table df720ae6-8978-47dc-92df-48a377e3bcdc: Starting a new Raft election for term 2.
2015-09-22T15:50:42.481147447 380.850175s error:

Tue Sep 22 15:50:38 2015

1: backtrace_t::backtrace_t() at backtrace.cc:203
2: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at basic_string.h:269
3: format_backtrace(bool) at backtrace.cc:197
4: report_fatal_error(char const*, int, char const*, ...) at basic_string.h:287
5: linux_thread_pool_t::fatal_signal_handler(int, siginfo*, void*) at thread_pool.cc:359
6: /lib/x86_64-linux-gnu/libpthread.so.0(+0xfcb0) [0x7f6c4a847cb0] at 0x7f6c4a847cb0 (/lib/x86_64-linux-gnu/libpthread.so.0)
7: primary_dispatcher_t::dispatchee_registration_t::mark_ready() at primary_dispatcher.cc:51
8: mailbox_manager_t::mailbox_read_coroutine(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, threadnum_t, unsigned long, std::vector<char, std::allocator<char> >*, long, mailbox_manager_t::force_yield_t) at mailbox.cc:277
9: /home/ssd3/daniel/rethinkdb/build/release_clang/rethinkdb() [0x777938] at 0x777938 ()
10: coro_t::run() at coroutines.cc:214
2015-09-22T15:50:42.484775028 380.853803s error: Exiting
```

**Comment by danielmewes** [comment]
This might just be a destruction order race in `remote_replicator_server_t`

**Comment by danielmewes** [comment]
The likely fix is in CR 3239

**Comment by danielmewes** [comment]
Fixed in next 7ed3be21e29fb53e9303b9b378c04f3c623fb010 and v2.1.x 8b4bba6
